js/adapt-stateful-session.js

```diff
@@ -61,29 +61,40 @@ define([
       var sessionPairs = Adapt.offlineStorage.get();
       var hasNoPairs = _.keys(sessionPairs).length === 0;
 
+      var courseState;
       var doSynchronousPart = function() {
-        if (sessionPairs.questions && this._shouldStoreResponses) questions.deserialize(sessionPairs.questions);
-        if (sessionPairs._isCourseComplete) Adapt.course.set('_isComplete', sessionPairs._isCourseComplete);
-        if (sessionPairs._isAssessmentPassed) Adapt.course.set('_isAssessmentPassed', sessionPairs._isAssessmentPassed);
+        if (sessionPairs.q && this._shouldStoreResponses) questions.deserialize(sessionPairs.q);
+        if (courseState) Adapt.course.set('_isComplete', courseState[0]);
+        if (courseState) Adapt.course.set('_isAssessmentPassed', courseState[1]);
         callback();
       }.bind(this);
 
       if (hasNoPairs) return callback();
 
+      if (sessionPairs.c) {
+        courseState = SCORMSuspendData.deserialize(sessionPairs.c);
+      }
+
       // Asynchronously restore block completion data because this has been known to be a choke-point resulting in IE8 freezes
-      if (sessionPairs.completion) {
-        serializer.deserialize(sessionPairs.completion, doSynchronousPart);
+      if (courseState) {
+        serializer.deserialize(courseState.slice(2).map(Number).map(String).join(''), doSynchronousPart);
       } else {
         doSynchronousPart();
       }
     },
 
     getSessionState: function() {
+      var blockCompletion = serializer.serialize();
+      var courseComplete = Adapt.course.get('_isComplete') || false;
+      var assessmentPassed = Adapt.course.get('_isAssessmentPassed') || false;
+      Adapt.log.info(`course._isComplete: ${courseComplete}, course._isAssessmentPassed: ${assessmentPassed}, BlockCompletion: ${blockCompletion}`);
+      var courseState = SCORMSuspendData.serialize([
+        courseComplete,
+        assessmentPassed,
+      ].concat(blockCompletion.split('').map(Number).map(Boolean)));
```

Comment:
Spread operator feels very readable here:
```suggestion
        ...blockCompletion.split('').map(Number).map(Boolean)
      ]);

Reply:
Did you check that it works with the compiler?